ILM: Moving full inspection of all indices to MasterFailOver condition

PDTCCLF opened the pull request:
This is my attempt at solving issue #80407. As per the description of the issue, the costly loop that inspects all indices of a cluster state is no longer being executed by default. A full inspection only happens when an exception occurs to the master, inspired by SnapshotService.

PDTCCLF committed:
Update IndexLifecycleService.java

PDTCCLF committed:
Add onMasterFailOver

PDTCCLF committed:
Merge branch 'master' into master

PDTCCLF committed:
Update IndexLifecycleService.java

elasticmachine commented:
Pinging @elastic/es-data-management (Team:Data Management)

PDTCCLF committed:
Create 87616.yaml

PDTCCLF committed:
Remove capitalize and add import

PDTCCLF committed:
Revert "Remove capitalize and add import"

This reverts commit 953f24202559b7524c7bd386bf1c911a4ec8f9a8.

PDTCCLF committed:
Fix missing brace

joegallo commented on x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/ilm/IndexLifecycleService.java:
```diff
@@ -208,9 +212,9 @@ void onMaster(ClusterState clusterState) {
                         if (logger.isTraceEnabled()) {
                             logger.warn(
-                                () -> format(
+                                new ParameterizedMessage(
```
Thise lambda to `ParameterizedMessage` look like a bad merge to me, please undo it.

joegallo commented on x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/ilm/IndexLifecycleService.java:
```diff
@@ -208,9 +212,9 @@ void onMaster(ClusterState clusterState) {
+                                new ParameterizedMessage(
                                     "async action execution failed during master election trigger"
-                                        + " for index [%s] with policy [%s] in step [%s], lifecycle state: [%s]",
+                                        + " for index [{}] with policy [{}] in step [{}], lifecycle state: [{}]",
```
Same.

joegallo commented on x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/ilm/IndexLifecycleService.java:
```diff
@@ -220,9 +224,9 @@ void onMaster(ClusterState clusterState) {
                         } else {
                             logger.warn(
-                                () -> format(
+                                new ParameterizedMessage(
```
Same.

joegallo commented on x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/ilm/IndexLifecycleService.java:
```diff
@@ -243,6 +247,7 @@ void onMaster(ClusterState clusterState) {
         }
     }
 
+
```
Please remove errant whitespace change.

joegallo reviewed (changes requested):
Beyond my other issues, the code associated with this PR doesn't compile. That's a pretty big red flag.

PDTCCLF commented:
> Beyond my other issues, the code associated with this PR doesn't compile. That's a pretty big red flag.

I see. I will try to fix the details you mentioned and make sure it does compile. I'm sorry, I believed it had passed gradle check, but I was mistaken.

joegallo commented:
No worries, best of luck.

PDTCCLF committed:
Removing whitespace, bad merge and critical flaws

The previous version of the PR did not compile due to a variable that was being called on MasterFailOver, but was not set. I fixed this problem, plus the mentioned errant whitespace and bad merge.

gmarouli commented:
Hi @PDTCCLF, thank you for working on this. I see there hasn't been any progress in this PR since July, I hope you do not mind if I give it a try myself. Let me know if you are still actively working on it. Then I will look for something else.

PDTCCLF commented:
> Hi @PDTCCLF, thank you for working on this. I see there hasn't been any progress in this PR since July, I hope you do not mind if I give it a try myself. Let me know if you are still actively working on it. Then I will look for something else. 

Not at all. You can work on it.  Thanks for asking.

elasticsearchmachine commented:
Pinging @elastic/es-storage-engine (Team:StorageEngine)
